dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileHandler.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayDeque;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ * Handler for Linkage Checker exclusion XML files.
+ *
+ * <p>An instance of this class can process one document and cannot be reused for other documents.
+ */
+class ExclusionFileHandler extends DefaultHandler {
+
+  /**
+   * Stack to hold the ancestors of the XML element in {@link #startElement(String, String, String,
+   * Attributes)}.
+   */
+  private ArrayDeque<SymbolProblemMatcher> stack = new ArrayDeque<>();
+
+  private ImmutableList.Builder<LinkageErrorMatcher> matchers;
+
+  ImmutableList<LinkageErrorMatcher> getMatchers() {
+    return matchers.build();
+  }
+
+  @Override
+  public void startDocument() {
+    if (matchers != null) {
+      throw new IllegalStateException("This handler started reading document already");
+    }
+    matchers = ImmutableList.builder();
+  }
+
+  private void addMatcherToTop(Object child) throws SAXException {
```

Comment:
Why is the argument type Object? It seems this could be more specific.

Reply:
Good catch. It's SymbolProblemTargetMatcher interface. Updated.